modules/siddhi-extensions/markov-models/src/main/java/org/wso2/siddhi/extension/markov/MarkovChainStreamProcessor.java

```diff
@@ -0,0 +1,280 @@
+        } else {
+            throw new ExecutionPlanValidationException(
+                    "Alert threshold probability should be of type double. But found "
+                            + attributeExpressionExecutors[3].getReturnType());
+        }
+
+        Object object = attributeExpressionExecutors[4].execute(null);
+        if (object instanceof String) {
+            markovMatrixStorageLocation = (String) object;
+            trainingMode = TrainingMode.PREDEFINED_MATRIX;
+
+            File file = new File(markovMatrixStorageLocation);
+            if (!file.exists()) {
+                throw new ExecutionPlanValidationException(
+                        markovMatrixStorageLocation + " is not exists. Please provide a valid file path");
+            } else if (!file.isFile()) {
+                throw new ExecutionPlanValidationException(
+                        markovMatrixStorageLocation + " is not a file. Please provide a valid csv file");
+            }
+
+        } else if (object instanceof Integer) {
+            notificationsHoldLimit = (Integer) object;
+        } else if (object instanceof Long) {
+            notificationsHoldLimit = (Long) object;
+        } else {
+            throw new ExecutionPlanValidationException(
+                    "5th parameter should be the Training batch size or Markov matrix storage location. "
+                            + "They should be of types int/long or String. But found "
+                            + attributeExpressionExecutors[4].getReturnType());
+        }
+
+        if (attributeExpressionExecutors.length == 6) {
+
+            if (attributeExpressionExecutors[5] instanceof ConstantExpressionExecutor) {
+                Object trainingOptionObject = attributeExpressionExecutors[5].execute(null);
+                if (trainingOptionObject instanceof Boolean) {
+                    trainingOption = (Boolean) trainingOptionObject;
+                } else {
+                    throw new ExecutionPlanValidationException("Training option should be of type boolean. But found "
+                            + attributeExpressionExecutors[5].getReturnType());
+                }
+            } else {
+                trainingOptionExpressionExecutor = attributeExpressionExecutors[5];
+            }
+
+        }
+
+        if (trainingMode == TrainingMode.PREDEFINED_MATRIX) {
+            markovChainTransitionProbabilitiesCalculator = new MarkovChainTransitionProbabilitiesCalculator(
+                    durationToKeep, alertThresholdProbability, markovMatrixStorageLocation);
+        } else {
+            markovChainTransitionProbabilitiesCalculator = new MarkovChainTransitionProbabilitiesCalculator(
+                    durationToKeep, alertThresholdProbability, notificationsHoldLimit);
+
+        }
+
+        List<Attribute> attributeList = new ArrayList<Attribute>(3);
+        attributeList.add(new Attribute("lastState", Attribute.Type.STRING));
```

Comment:
Most probably this "lastState" etc, will be used at somewhere else. Better to put them as some public constants in relevant class.